pandas/core/indexes/base.py

```diff
@@ -785,7 +785,13 @@ def copy(self, name=None, deep=False, dtype=None, **kwargs):
             new_index = self._shallow_copy()
 
         names = kwargs.get('names')
-        names = self._validate_names(name=name, names=names, deep=deep)
+
+        if name is None and names is None:
+            from copy import deepcopy
```

Comment:
import can be at the top. I think write like this
```
names = self._validate_names(name=name, names=names)
names = deepcopy(self.names) if deep else self.names
```

to avoid the double check here